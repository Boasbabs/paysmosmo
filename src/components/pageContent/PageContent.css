.page-content--container {
  background-color: #fff;
  padding: 10px 50px;
  margin-bottom: 2em;
}

.page-content--header {
  background-color: #27ae60;
  color: #fff;
}
.page-content--header .ant-page-header-title-view-title {
  color: #fff;
  text-transform: uppercase;
}
.side-menu--shadow {
  padding: 0 !important;
  box-shadow: -1px 7px 8px 0px rgba(0, 0, 0, 0.19),
    0px 4px 3px -12px rgba(0, 0, 0, 0.19);
}

.ant-carousel .slick-slide {
  text-align: center;
  height: 410px;
  line-height: 160px;
  background: #364d79;
  overflow: hidden;
}

.banner-image {
  object-fit: cover;
  background-size: cover;
  background-repeat: no-repeat;
  background-position: 50% 50%;
  height: 100%;
  width: 100%;
}

.carousel {
  background: linear-gradient(rgba(0, 0, 0, 0.6), rgba(0, 0, 0, 0.6)),
    url(../images/landing-image.jpg);
  background-size: cover;
  width: 100%;
  height: 410px;
  position: relative;
  color: #fff;
  text-align: left;
  /* float: left; */
  line-height: 20px;
}

.carousel--header {
  text-transform: uppercase;
  letter-spacing: 2px;
  margin: 1em 0;
  font-size: 32px;
  font-weight: bolder;
  position: absolute;
  top: 30%;
  left: 10%;
}

.carousel--text {
  font-weight: bold;
  color: white;
  line-height: 22px;
  /* letter-spacing: 1px; */
  position: absolute;
  top: 50%;
  left: 10%;
  width: 52%;
}
.carousel--btn {
  background-color: transparent;
  border: 2px #f1d51c solid;
  color: #f1d51c;
  font-weight: bold;
  left: 10%;
  margin: 2em 0;
  position: absolute;
  top: 55%;
  text-transform: uppercase;
}

.carousel--btn:hover {
  background-color: #f1d51c;
  color: #fff;
  border: 2px #f1d51c solid;
}

@media screen and (max-width: 992px) {
  .carousel--header {
    margin: 0;
    font-size: 22px;
    font-weight: bolder;
    position: absolute;
    top: 35%;
  }

  .carousel--text {
    font-size: 12px;
    line-height: 1;
    position: absolute;
    top: 50%;
    width: 75%;
  }
}
