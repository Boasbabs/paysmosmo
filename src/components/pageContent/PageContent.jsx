import React, { Component } from "react";
import {
  Layout,
  Row,
  Col,
  Menu,
  Carousel,
  PageHeader,
  Button,
  Spin
} from "antd";
import axios from "axios";
import { CATEGORY_ENDPOINT, ACCESS_TOKEN } from "../constants";
import "./PageContent.css";

const { Content } = Layout;
const SubMenu = Menu.SubMenu;

//axios.defaults.headers.common['Authorization'] = ACCESS_TOKEN;

class PageContent extends Component {
  state = { categoryData: [] };

  componentDidMount() {
    // make API request
    axios
      .get(CATEGORY_ENDPOINT, { headers: { Authorization: ACCESS_TOKEN } })
      .then(response => {
        // handle success
        const categoryData = response.data;
        this.setState({ categoryData });
      })
      .catch(error => {
        // handle error
        console.log(error);
      });
  }
  render() {
    // destruction categoryData from state
    const { categoryData } = this.state;
    return (
      <React.Fragment>
        <Content className="page-content--container">
          <Row gutter={24}>
            <Col
              xs={24}
              sm={24}
              md={12}
              lg={7}
              xl={7}
              className="side-menu--shadow"
            >
              <PageHeader title="Categories" className="page-content--header" />
              <Menu style={{ width: "100%", padding: 0 }} mode="vertical">
                {/* Iterate over the categoryData using Js Array map method */}
                { categoryData.length === 0 ? (
                  // Display spinner while data is fetch
                  // <div className="loading-container">
                    <Spin size="large" tip="Loading..." />
                  // </div>
                ) : (
                  categoryData.map(({ name, id }) => (
                    <SubMenu key={id} title={<span>{name}</span>}>
                      <Menu.Item key="1">Option 1</Menu.Item>
                      <Menu.Item key="2">Option 2</Menu.Item>
                    </SubMenu>
                  ))
                )}
              </Menu>
            </Col>
            <Col xs={24} sm={24} md={12} lg={17} xl={17}>
              <Carousel autoplay className="">
                <div className="carousel ">
                  <p className="carousel--header">New Collection 1</p>
                  <p className="carousel--text">
                    It is a long established fact that a reader will be
                    distracted by the readable content of a page when looking at
                    its layout.
                  </p>
                  <Button className="carousel--btn" size="large">
                    Shop Now
                  </Button>
                </div>
                <div className="carousel ">
                  <p className="carousel--header">New Collection 2</p>
                  <p className="carousel--text">
                    It is a long established fact that a reader will be
                    distracted by the readable content of a page when looking at
                    its layout.
                  </p>
                  <Button className="carousel--btn" size="large">
                    Shop Now
                  </Button>
                </div>
                <div className="carousel ">
                  <p className="carousel--header">New Collection 3</p>
                  <p className="carousel--text">
                    It is a long established fact that a reader will be
                    distracted by the readable content of a page when looking at
                    its layout.
                  </p>
                  <Button className="carousel--btn" size="large">
                    Shop Now
                  </Button>
                </div>
              </Carousel>
            </Col>
          </Row>
        </Content>
      </React.Fragment>
    );
  }
}

export default PageContent;
