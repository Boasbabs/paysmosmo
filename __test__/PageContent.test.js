import React from "react";
import { create } from "react-test-renderer";
import PageContent from "../src/components/pageContent";

describe("PageContent component", () => {
  test("it matches snapshot", () => {
    const component = create(<PageContent />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
