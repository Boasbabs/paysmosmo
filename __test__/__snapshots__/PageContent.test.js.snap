// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent component it matches snapshot 1`] = `
<main
  className="page-content--container ant-layout-content"
>
  <div
    className="ant-row"
    style={
      Object {
        "marginLeft": -12,
        "marginRight": -12,
      }
    }
  >
    <div
      className="side-menu--shadow ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-7 ant-col-xl-7"
      style={
        Object {
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      <div
        className="ant-page-header page-content--header"
      >
        <div
          className="ant-page-header-title-view"
        >
          <span
            className="ant-page-header-title-view-title"
          >
            Categories
          </span>
        </div>
      </div>
      <ul
        className="ant-menu ant-menu-light ant-menu-root ant-menu-vertical"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={
          Object {
            "padding": 0,
            "width": "100%",
          }
        }
      >
        <li
          className="ant-menu-item loading-container"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={Object {}}
        >
          <div
            className="ant-spin ant-spin-lg ant-spin-spinning ant-spin-show-text"
          >
            <span
              className="ant-spin-dot ant-spin-dot-spin"
            >
              <i />
              <i />
              <i />
              <i />
            </span>
            <div
              className="ant-spin-text"
            >
              Loading...
            </div>
          </div>
        </li>
      </ul>
    </div>
    <div
      className="ant-col-xs-24 ant-col-sm-24 ant-col-md-12 ant-col-lg-17 ant-col-xl-17"
      style={
        Object {
          "paddingLeft": 12,
          "paddingRight": 12,
        }
      }
    >
      <div
        className="ant-carousel"
      >
        <div
          className="slick-slider slick-initialized"
          dir="ltr"
        >
          <div
            className="slick-list"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={null}
            onMouseMove={null}
            onMouseUp={[Function]}
            onTouchCancel={null}
            onTouchEnd={[Function]}
            onTouchMove={null}
            onTouchStart={[Function]}
            style={Object {}}
          >
            <div
              className="slick-track"
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseOver={[Function]}
              style={
                Object {
                  "WebkitTransform": "translate3d(0px, 0px, 0px)",
                  "WebkitTransition": "",
                  "msTransform": "translateX(0px)",
                  "opacity": 1,
                  "transform": "translate3d(0px, 0px, 0px)",
                  "transition": "",
                }
              }
            >
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={-1}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 3
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={false}
                className="slick-slide slick-active slick-current"
                data-index={0}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 1
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide"
                data-index={1}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 2
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide"
                data-index={2}
                onClick={[Function]}
                style={
                  Object {
                    "outline": "none",
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 3
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={3}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 1
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={4}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 2
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                aria-hidden={true}
                className="slick-slide slick-cloned"
                data-index={5}
                onClick={[Function]}
                style={
                  Object {
                    "width": 0,
                  }
                }
                tabIndex="-1"
              >
                <div>
                  <div
                    className="carousel "
                    style={
                      Object {
                        "display": "inline-block",
                        "width": "100%",
                      }
                    }
                    tabIndex={-1}
                  >
                    <p
                      className="carousel--header"
                    >
                      New Collection 3
                    </p>
                    <p
                      className="carousel--text"
                    >
                      It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout.
                    </p>
                    <button
                      className="ant-btn carousel--btn ant-btn-lg"
                      onClick={[Function]}
                      type="button"
                    >
                      <span>
                        Shop Now
                      </span>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <ul
            className="slick-dots"
            onMouseEnter={null}
            onMouseLeave={null}
            onMouseOver={null}
            style={
              Object {
                "display": "block",
              }
            }
          >
            <li
              className="slick-active"
            >
              <button
                onClick={[Function]}
              >
                1
              </button>
            </li>
            <li
              className=""
            >
              <button
                onClick={[Function]}
              >
                2
              </button>
            </li>
            <li
              className=""
            >
              <button
                onClick={[Function]}
              >
                3
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</main>
`;
